run.py

```diff
@@ -144,34 +143,46 @@ def cleanup(log, dicom_directory, output_dir):
     # This prevents the script from coppying in subdirectories to the output.  I don't
     # Think this particular gear does that anyway, though.
     for file_name in src_files:
+
         full_file_name = os.path.join(dicom_directory, file_name)
+
         if os.path.isfile(full_file_name):
-            dest = output_dir / Path(file_name)
+
+            if Path(file_name).suffix == '.dcm':
+                if not os.path.exists(dicom_directory / 'dcms_to_zip'):
+                    os.mkdir(dicom_directory / 'dcms_to_zip')
+                dest = dicom_directory / 'dcms_to_zip' / file_name
+
+            else:
+                dest = output_dir / Path(file_name)
+
             os.rename(full_file_name, dest)
 
+    col = DICOMCollection.from_dir(dicom_directory / Path('dcms_to_zip'))
+    col.to_zip(output_dir / Path(f'{Path(context.get_input_path("Input_file")).stem}.dicom.zip'))
```

Comment:
where is `context` coming from actually ? (not an arg of your function)

Also you can remove Path in these lines as:
```
    col = DICOMCollection.from_dir(dicom_directory / 'dcms_to_zip')
    col.to_zip(output_dir / f'{Path(context.get_input_path("Input_file")).stem}.dicom.zip')
```

Reply:
It did run w/out context added, so maybe is it global to that function? In any case added it as an argument.